spec/index.bs

```diff
@@ -6516,108 +6552,35 @@ GPUQueue includes GPUObjectBase;
-};
-</script>
-
-<dl dfn-type=method dfn-for=GPUQueue>
-    : <dfn>createFence(descriptor)</dfn>
+    : <dfn>wait(queue, value)</dfn>
     ::
-        Creates a {{GPUFence}}.
+        Waits for the |queue| reaching |value| on the [=Queue timeline=]. If |value| is undefined,
+        the |queue|.{{GPUQueue/lastSignaledValue}} is used instead.
 
-        <div algorithm="GPUQueue.createFence">
-            **Called on:** {{GPUQueue}} this.
+        All the resource transitions from |queue| targeting |this| will be considered done on the [=Queue timeline=]
+        after this call, and relevant resources can be used on |this|.
 
-            **Arguments:**
-            <pre class=argumentdef for="GPUQueue/createFence(descriptor)">
-                descriptor: Description of the {{GPUFence}} to create.
-            </pre>
-
-            **Returns:** {{GPUFence}}
-
-            Issue: Describe {{GPUQueue/createFence()}} algorithm steps.
-        </div>
-</dl>
-
-### Completion ### {#fence-signaling}
-
-Completion of a fence is signaled from the {{GPUQueue}} that created it.
-
-<dl dfn-type=method dfn-for=GPUQueue>
-    : <dfn>signal(fence, signalValue)</dfn>
+    : <dfn>signal()</dfn>
     ::
-        Signals the given fence, increasing it's completed value to `signalValue`.
+        Returns a signal value corresponding to the work submitted to the queue up to this moment on the [=content timeline=].
+        This value has to be higher or equal to return value of the previous call to {{GPUQueue/signal()}} on this queue.
 
-        <div algorithm="GPUQueue.signal">
-            **Called on:** {{GPUQueue}} this.
-
-            **Arguments:**
-            <pre class=argumentdef for="GPUQueue/signal(fence, signalValue)">
-                |fence|: The fence to signal.
-                signalValue: The value to increase |fence|'s completion value to.
-            </pre>
-
-            **Returns:** {{undefined}}
-
-            Issue: Describe {{GPUQueue/signal()}} algorithm steps.
-        </div>
-</dl>
-
-The completion of the fence and the value it completes with can be observed from the {{GPUFence}}.
-
-<dl dfn-type=method dfn-for=GPUFence>
-    : <dfn>getCompletedValue()</dfn>
-    ::
-        Returns the largest signalled value completion value for the fence that has propagated to
-        the [=content timeline=].
-
-        <div algorithm="GPUFence.getCompletedValue">
-            **Called on:** {{GPUFence}} this.
-
-            **Returns:** {{GPUFenceValue}}
-
-            Issue: Describe {{GPUFence/getCompletedValue()}} algorithm steps.
-        </div>
-
-    : <dfn>onCompletion(completionValue)</dfn>
+    : <dfn>onCompletion(signalValue)</dfn>
     ::
-        Returns a {{Promise}} that resolves once the fence's completion value &ge; `completionValue`.
+        Returns a {{Promise}} that resolves once the queue reaches `signalValue`
+        (or |this|.{{GPUQueue/lastSignaledValue}}) if undefined) on the [=content timeline=].
```

Comment:
I think this is better as requiring `onCompletion(this.lastSignaledValue)`. The implicit version `onCompletion()` looks like it'll resolve when all pending queue operations have completed.

```
q.submit(A)
const v = q.signal()
q.submit(B)
await q.onCompletion()
```

Users would naively expect this to resolve after `q.submit(B)` is complete.
The explicit version of what those users want is simple and clear: `q.onCompletion(q.signal())`

Reply:
Makes sense!